From hackermail Thu Jan  1 00:00:00 1970
From: SeongJae Park <sj@kernel.org>
To: Andrew Morton <akpm@linux-foundation.org>
Cc: SeongJae Park <sj@kernel.org>, "# 6 . 17-rc1" <stable@vger.kernel.org>, damon@lists.linux.dev, kernel-team@meta.com, linux-kernel@vger.kernel.org, linux-mm@kvack.org
Message-Id: <20250909022238.2989-2-sj@kernel.org>
In-Reply-To: <20250909022238.2989-1-sj@kernel.org>
Date: Mon,  8 Sep 2025 19:22:36 -0700
Subject: [PATCH 1/3] samples/damon/wsse: avoid starting DAMON before initialization

Commit 0ed1165c3727 ("samples/damon/wsse: fix boot time enable
handling") is somehow incompletely applying the origin patch [1].  It is
missing the part that avoids starting DAMON before module
initialization.  Probably a mistake during a merge has happened.  Fix it
by applying the missed part again.

[1] https://lore.kernel.org/20250706193207.39810-2-sj@kernel.org

Fixes: 0ed1165c3727 ("samples/damon/wsse: fix boot time enable handling")
Cc: <stable@vger.kernel.org> # 6.17-rc1
Signed-off-by: SeongJae Park <sj@kernel.org>
Link: https://patch.msgid.link/20250909022238.2989-2-sj@kernel.org
Cc: "# 6 . 17-rc1" <stable@vger.kernel.org>
---
 samples/damon/wsse.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/samples/damon/wsse.c b/samples/damon/wsse.c
index da052023b099..21eaf15f987d 100644
--- a/samples/damon/wsse.c
+++ b/samples/damon/wsse.c
@@ -118,6 +118,9 @@ static int damon_sample_wsse_enable_store(
 		return 0;
 
 	if (enabled) {
+		if (!init_called)
+			return 0;
+
 		err = damon_sample_wsse_start();
 		if (err)
 			enabled = false;
-- 
2.39.5